mobify/progressive-web-scaffold
# Ignore SVG imports in Jest tests

With no loader to handle SVG imports, Jest tests fail with a SyntaxError:

<img width="489" alt="screen shot 2016-10-04 at 2 10 32 pm" src="https://cloud.githubusercontent.com/assets/1211726/19092776/26557730-8a3d-11e6-82a0-159a556fc83a.png">

## Changes
- Add a [moduleNameMapper](https://facebook.github.io/jest/docs/tutorial-webpack.html) alias for anything that has `.svg` extension
- Create a mock file to load instead of SVGs
- Add an example test which imports the `<Icon>` SDK component, which itself imports an SVG via `<IconSprite>`
- Simplify the regEx used for finding test files in the project
- Remove `ignore-styles` imports in test files (as they serve no purpose here - aligns with change to component generator in SDK)

## How to test-drive this PR
- Remove the `moduleNameMapper` object from `package.json`
- Run `npm run test -- --no-cache`
- Verify test fails
- Revert change to `package.json` and repeat steps
- Verify test passes
